xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -274,6 +275,40 @@ public static String escapeAttributeValue(Object content)
         return result.toString();
     }
 
+    /**
+     *  Escapes some XML special characters (&amp; and &lt;) in a <code>String</code> using numerical XML
+     *  entities, so that the resulting string can safely be used in HTML pages.
+     *
+     * @param content the text to escape, may be {@code null}
+     * @return a new escaped {@code String}, {@code null} if {@code null} input
+     * @since 12.8RC1
+     */
+    @Unstable
+    public static String minimalEscape(Object content)
+    {
+        if (content == null) {
+            return null;
+        }
+        String str = String.valueOf(content);
+        StringBuilder result = new StringBuilder((int) (str.length() * 1.1));
+        int length = str.length();
```

Comment:
You're using ``str.length()`` above also so I would declare the variable before.

Reply:
Factored here https://github.com/xwiki/xwiki-commons/pull/101/files#diff-c1eed56adbad8fd67a8e196907ce032fR295